### PR TITLE
[docs] #85: Add UML diagrams, create a new chapter about objects

### PR DESCRIPTION
**Pull request by outoftardis**
Signed-off-by: Ekaterina Mekhnetsova <mekkatya@gmail.com>

- a new chapter that covers the info about objects in a blockchain
- UML diagrams with the hierarchy (I've only added a couple, not covering everything in the hierarchy yet. This is just to make sure this is the right format)
- the topic on metadata from the advanced section was moved to this new chapter
- a separate topic with the info about assets was created within this same chapter (that info used to be copy-pasted between lang-specific guides)
- minor changes in the topic on permissions

### CODE REVIEW

File: src/guide/diagrams-src/domain-account-asset.puml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+
+AssetDefinition : id
+AssetDefinition : value_type
+AssetDefinition : mintable
+AssetDefinition : metadata
+
+Asset : id
+Asset : value
+@enduml
```
Comment (appetrosyan) [inline src/guide/diagrams-src/domain-account-asset.puml:34]: Consider adding newlines at end of file. This isn't strictly necessary, but every other file has a newline at the end.

**Review by appetrosyan — APPROVED** [review]

File: src/guide/advanced/permissions.md [hunk 2]
```diff
@@ -338,15 +340,18 @@ let grant = Instruction::Grant(GrantBox::new(
 
 ## Permission Groups (Roles)
 
-A set of permissions is called a **role**. Roles can be granted and
-revoked.
+A set of permissions is called a **role**. Similarly to permission tokens,
+roles can be granted using `Grant` instruction and revoked using `Revoke`
```
Comment (WRRicht3r) [inline src/guide/advanced/permissions.md:344]:
```suggestion
roles can be granted using. the `Grant` instruction and revoked using the `Revoke`
```

File: src/guide/advanced/permissions.md [hunk 3]
```diff
@@ -338,15 +340,18 @@ let grant = Instruction::Grant(GrantBox::new(
...
+instruction.
+
+Before granting a role to an account, a role should first be registered.
```
Comment (WRRicht3r) [inline src/guide/advanced/permissions.md:347]:
```suggestion
Before granting a role to an account, the role should be registered first.
```

File: src/guide/advanced/queries.md [hunk 4]
```diff
@@ -116,8 +116,9 @@ query results.
 
 ::: stable
 
-Sorting only applies to entities that have [metadata](./metadata.md) as the
-metadata key is used to sort query results.
+Sorting only applies to entities that have
+[metadata](../objects/metadata.md) as the metadata key is used to sort
```
Comment (WRRicht3r) [inline src/guide/advanced/queries.md:120]:
```suggestion
[metadata](../objects/metadata.md), as the metadata key is used to sort
```

File: src/guide/objects/hierarchy.md [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+The diagram below provides a more detailed illustration of the relationship
+between domains, accounts, and assets in the blockchain. You can learn more
+about [permissions and roles](../advanced/permissions.md) and
+[metadata](metadata.md) in the corresponding sections of the tutorial. The
+asset structure is illustrated in a [dedicated chapter](./assets.md).
+
+::: info
+
+Note that the diagram below depicts Account structure for the LTS version
+of Iroha. In the stable version, permission tokens are stored not within
```
Comment (WRRicht3r) [inline src/guide/objects/hierarchy.md:35]:
```suggestion
of Iroha. In the stable version, permission tokens aren't stored within
```

File: src/guide/objects/metadata.md [hunk 6]
```diff
@@ -58,8 +58,8 @@ let set_hat_color = SetKeyValueBox::new(
 ## Working with metadata
 
 The following example showcases how to register and grant a
-[role](./permissions.md#permission-groups-roles) for accessing metadata of
-another account.
+[role](../advanced/permissions.md#permission-groups-roles) for accessing
```
Comment (WRRicht3r) [inline src/guide/objects/metadata.md:61]:
```suggestion
[role](../advanced/permissions.md#permission-groups-roles) for accessing the
```

**Review by WRRicht3r — CHANGES_REQUESTED** [review]

**Review by 6r1d — APPROVED** [review]
LGTM, ping me after addressing Willam's changes if GitHub requires a re-approval.

File: src/guide/advanced/permissions.md (hunk 2 above)
Comment (outoftardis) [reply src/guide/advanced/permissions.md:344]:
```suggestion
roles can be granted using the `Grant` instruction and revoked using the `Revoke`
```

**Review by WRRicht3r — APPROVED** [review]

**Pull request merged**